modules/mod-deep-learning/ModelManagerPanel.cpp

```diff
@@ -0,0 +1,336 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   ModelManagerPanel.cpp
+   Hugo Flores Garcia
+
+******************************************************************/
+
+#include "ModelManagerPanel.h"
+
+#include "DeepLearningEffectBase.h"
+#include "CodeConversions.h"
+#include "ExploreHuggingFaceDialog.h"
+#include "effects/EffectUI.h"
+
+#include "Shuttle.h"
+#include "ShuttleGui.h"
+
+#include <wx/display.h>
+#include <wx/scrolwin.h>
+#include <wx/range.h>
+#include <wx/gauge.h>
+#include <wx/button.h>
+#include <wx/sizer.h>
+#include <wx/textdlg.h>
+#include <wx/stattext.h>
+#include <wx/hyperlink.h>
+
+#include "widgets/AudacityTextEntryDialog.h"
+#include "Internat.h"
+#include "AllThemeResources.h"
+#include "Theme.h"
+
+// ModelManagerPanel
+ModelManagerPanel::ModelManagerPanel(wxWindow *parent, DeepLearningEffectBase *effect)
+                                    : wxPanelWrapper(parent)
+{
+   mEffect = effect;
+   mTools = nullptr;
+
+   ShuttleGui S(this, eIsCreating);
+   PopulateOrExchange(S);
+   Layout();
+   Fit();
+   Center();
+}
+
+void ModelManagerPanel::PopulateOrExchange(ShuttleGui & S)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+   S.StartVerticalLay(true);
+   {
+      mTools = safenew ManagerToolsPanel(S.GetParent(), this);
+      // mTools->PopulateOrExchange(S);
+      S.AddWindow(mTools);
+
+      S.StartMultiColumn(2, wxEXPAND);
+      {
+         mScroller = S.StartScroller(wxVSCROLL);
```

Comment:
Do I understand right that this scroller may get populated with stuff, and grow before your eyes, as a result of asynchronous activity of the ModelManager?  That is unusual, as our dialogs go, and is worth a comment!